controllers/bgppeer_controller_test.go

```diff
@@ -0,0 +1,544 @@
+						return "", err
+					}
+					return configmap.Data[consts.MetalLBConfigMapName], err
+				}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(`peers:
+- my-asn: 64500
+  node-selectors:
+  - match-expressions:
+    - key: kubernetes.io/hostname
+      operator: In
+      values:
+      - hostA hostB
+  peer-address: 10.0.0.1
+  peer-asn: 64501
+  router-id: 10.10.10.10
+
+- my-asn: 64000
+  peer-address: 11.0.0.1
+  peer-asn: 64001
+  router-id: 11.11.11.11
+
+`))
+			})
+
+			By("Deleting 1st BGPPeer resource", func() {
+				Peer := &v1alpha1.BGPPeer{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bgp-peer1",
+						Namespace: MetalLBTestNameSpace,
+					},
+					Spec: v1alpha1.BGPPeerSpec{
+						Address:  "10.0.0.1",
+						ASN:      64501,
+						MyASN:    64500,
+						RouterID: "10.10.10.10",
+						NodeSelectors: []v1alpha1.NodeSelector{
+							{
+								MatchExpressions: []v1alpha1.MatchExpression{
+									{
+										Key:      "kubernetes.io/hostname",
+										Operator: "In",
+										Values: []string{
+											"hostA",
+											"hostB",
+										},
+									},
+								},
+							},
+						},
+					},
+				}
+				err := k8sClient.Delete(context.Background(), Peer)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Checking ConfigMap is created
+				By("By checking ConfigMap matches bgp-peer2 configuration")
+				Eventually(func() (string, error) {
+					err := k8sClient.Get(context.Background(),
+						types.NamespacedName{Name: consts.MetalLBConfigMapName, Namespace: MetalLBTestNameSpace}, configmap)
+					if err != nil {
+						return "", err
+					}
+					return configmap.Data[consts.MetalLBConfigMapName], err
+				}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(`peers:
+- my-asn: 64000
+  peer-address: 11.0.0.1
+  peer-asn: 64001
+  router-id: 11.11.11.11
+`))
+			})
+			By("Deleting 2nd BGPPeer resource", func() {
+				Peer := &v1alpha1.BGPPeer{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bgp-peer2",
+						Namespace: MetalLBTestNameSpace,
+					},
+					Spec: v1alpha1.BGPPeerSpec{
+						Address:  "11.0.0.1",
+						ASN:      64001,
+						MyASN:    64000,
+						RouterID: "11.11.11.11",
+					},
+				}
+				err := k8sClient.Delete(context.Background(), Peer)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Checking ConfigMap is created
+				By("By checking ConfigMap deleted")
+				Eventually(func() (string, error) {
+					err := k8sClient.Get(context.Background(),
+						types.NamespacedName{Name: consts.MetalLBConfigMapName, Namespace: MetalLBTestNameSpace}, configmap)
+					if err != nil {
+						// if its notfound means that was the last bgppeer and configmap is deleted
+						if errors.IsNotFound(err) {
+							return "", nil
+						}
+						return "", err
+					}
+					return configmap.Data[consts.MetalLBConfigMapName], err
+				}, 2*time.Second, 200*time.Millisecond).Should(MatchYAML(``))
+			})
+		})
+	})
+
+	Context("Creating Full BGP configuration", func() {
```

Comment:
since there is a coupling between BGP and addresspool this is why I choose this file for the combined test cases , any new other controller should be using his own unit-test files and if the combined test start to grow we can have maybe one unit-test file for the combined test which will test multiple controllers